test/server/cards/01_SOR/events/OpenFire.spec.js

```diff
@@ -0,0 +1,56 @@
+describe('Open Fire', function() {
+    integration(function() {
+        describe('Open Fire\'s ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['open-fire'],
+                        groundArena: ['wampa'],
+                    },
+                    player2: {
+                        groundArena: ['pyke-sentinel', { card: 'fleet-lieutenant', upgrades: ['experience', 'experience'] }],
+                        spaceArena: [{ card: 'cartel-spacer', upgrades: ['shield'] }]
+                    }
+                });
+            });
+
+            // Base Damage is not allowed and thus is tested in the Select Exactly
+            // Similarly this should allow for units behind a sentinal to be selected
+
+            it('can damage a unit with a shield, removing only the shield', function () {
+                this.player1.clickCard(this.openFire);
+                expect(this.player1).toBeAbleToSelectExactly([this.wampa, this.cartelSpacer, this.pykeSentinel, this.fleetLieutenant]);
+
+                this.player1.clickCard(this.cartelSpacer);
+                expect(this.cartelSpacer).toHaveExactUpgradeNames([]);
+            });
+
+            it('can damage a unit without a shield, dealing damage to the unit with health to spare', function () {
+                this.player1.clickCard(this.openFire);
+                expect(this.player1).toBeAbleToSelectExactly([this.wampa, this.cartelSpacer, this.pykeSentinel, this.fleetLieutenant]);
+
+                this.player1.clickCard(this.wampa);
+                expect(this.wampa.damage).toBe(4);
+            });
+
+            it('can damage a damaged unit and destroy it', function () {
```

Comment:
In most contexts I'm all for super-thorough tests, in this case unfortunately we have a finite amount of Github Actions runner time and the jasmine tests are pretty heavy. So, we try to consolidate tests down as much as possible.

I think the first two tests have unique value and are good - we actually haven't written a test that checks shield blocking against direct damage effects yet, so the first one covers that case now.

The latter two tests I would prefer to cut, as both defeat from damage and stat bonuses due to upgrades are thoroughly tested elsewhere.

Reply:
That sounds reasonable :)